pkg/rtc/participant.go

```diff
@@ -92,6 +78,7 @@ type ParticipantParams struct {
 	ReconnectOnPublicationError  bool
 	ReconnectOnSubscriptionError bool
 	VersionGenerator             utils2.TimedVersionGenerator
+	TrackResolver                types.MediaTrackResolver
```

Comment:
creating an interface for this to delegate responsibility of finding the track and permission checking to Room